profiler/src/ProfilerEngine/Datadog.Linux.ApiWrapper/functions_to_wrap.c

```diff
@@ -24,15 +24,39 @@ The hijacked thread will start walking its callstack and a call to dl_iterate_ph
 dl_iterate_phdr is not recursive and the thread is blocked.
 
 %%%%%%%%%%%%%%%%%%%%%%%%%%% Fix
+One way is to use the LD_PRELOAD trick:
 We will rely on LD_PRELOAD mechanism to inject our own implementation of dl_iterate_phdr.
 Before calling the real dl_iterate_phdr, we block all signals that would intefere with the thread.
 Then will call the real dl_iterate_phdr.
-When finished, we put back the previous block signals.
+When finished, we put back the previous block signals. (This is done by libunwind, just taking advantage of it.)
 
-This is done by libunwind, just taking advantage of it.
+But this has a non-negligible overhead. Instead, we will incr/decr a counter per function
+each time the thread enters/exits from it.
+The profiler will just have to check if this counter is equal to 0 to profiler or not.
 
 */
 
+enum FUNCTION_ID
+{
+    ENTERED_DL_ITERATE_PHDR = 0,
+    ENTERED_DL_OPEN = 1,
+    ENTERED_DL_ADDR = 2,
+    ENTERED_PTHREAD_CREATE = 3,
+    ENTERED_PTHREAD_ATTR_INIT = 4,
+    ENTERED_PTHREAD_GETATTR_DEFAULT_NP = 5,
+    ENTERED_PTHREAD_SETATTR_DEFAULT_NP = 6,
+    ENTERED_FORK = 7
+};
+
+// counters: one byte per function
+__thread unsigned long long functions_entered_counter = 0;
+
+// this function is called by the profiler
+unsigned long long dd_can_be_profiled()
```

Comment:
I think the name is confusing because we can profile if it returns 0 (which is the opposite of what we would expect)

Reply:
I'll rename it to `dd_inside_wrapped_functions`.